src/Servers/Kestrel/Core/src/KestrelServerOptions.cs

```diff
@@ -121,9 +142,20 @@ public void ConfigureEndpointDefaults(Action<ListenOptions> configureOptions)
             EndpointDefaults = configureOptions ?? throw new ArgumentNullException(nameof(configureOptions));
         }
 
+        internal Func<string, Encoding?> GetRequestHeaderEncodingSelector()
+        {
+            if (ReferenceEquals(RequestHeaderEncodingSelector, DefaultRequestHeaderEncodingSelector) && Latin1RequestHeaders)
```

Comment:
Do we really need to keep the latin1 config flag? It was a temporary workaround for a specific partner. It's a clearer story if we remove it.

Reply:
The latin1 config flag is not documented and the RequestHeaderEncodingSelector will be, so I doubt there will be much confusion. Keeping back-compat is easy enough that I think we can support it indefinitely.

If it does cause problems in the future we can just remove it then. We could also consider logging a deprecation warning at startup if the latin1 config flag is set in .NET 5 and remove it in 6, but if the plan is just to immediately remove it, we could do that in any release.

Reply:
As a rule we remove all servicing tweaks in the next major version. This one is at about the same level as the AppContext switches.

Said partner didn't even want it to be a config switch, they set it via code. We can check with them, but I don't think we have a reason to keep this now that it's been replaced.

Reply:
There's a chance more than one partner is using this by now. We haven't been super tight lipped about the option and it's highly requested.

I also kinda like the config flag. If you're going to change the default encoding for all request headers, latin1 would be the encoding you want. The callback is more powerful, but what do you think about just documenting the Latin1RequestHeaders config switch the same time we doc the callback?

Reply:
Any more feedback on this PR @Tratcher? I don't think the config flag is like an AppContext switch because it's very similar to documented settings we read from config.

I understand not wanting to leave it there undocumented. Are you fine leaving it given we document the config flag the same time we document the new API?

Reply:
The work we did in the 3.1 patch had a lot of constraints that wouldn't apply in a normal release. If we reframe this as a new feature, would we still do it this way?
(And the person we should be asking about a new feature is @shirhatti, if he's fine with Latin1RequestHeaders as a permanent feature then I'll desist).

Requirements: Be able to specify the request header encoding from config.

Starting with that general premise I'd expect something more like: `"RequestHeaderEncoding": "latin1"`

 And that would more closely mirror the RequestHeaderEncodingSelector approach used in code that returns an encoder.

Reply:
I don't know that the config needs to be that flexible. But given there's room for debate and this was not originally designed to be a long-term thing, I removed it like you originally asked.

If @shirhatti thinks it's worth keeping, I can easily revert the commit that removes the flag in this PR.